sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -421,6 +425,10 @@ boolean isContentResponseOnWriteEnabled() {
      * @return a {@link CosmosPagedFlux} containing one or several feed response pages of read databases or an error.
      */
     public CosmosPagedFlux<CosmosDatabaseProperties> queryDatabases(SqlQuerySpec querySpec, CosmosQueryRequestOptions options) {
+        if (options == null) {
+            options = new CosmosQueryRequestOptions();
+        }
+
         return queryDatabasesInternal(querySpec, options);
```

Comment:
I thought that too. We are following this explicit check on api entry point in crud operations. So one thing is consistency, second we have github item https://github.com/Azure/azure-sdk-for-java/issues/13031 , where we will refactor all apis in one work item,  otherwise it will be confusing